distributed/dashboard/tests/test_components.py

```diff
@@ -25,9 +25,9 @@ def test_basic(Component):
 async def test_profile_plot(c, s, a, b):
     p = ProfilePlot()
     assert not p.source.data["left"]
-    await c.gather(c.map(slowinc, range(10), delay=0.05))
-    p.update(a.profile_recent)
-    assert len(p.source.data["left"]) >= 1
+    while not len(p.source.data["left"]):
+        await c.submit(slowinc, 1, pure=False, delay=0.1)
```

Comment:
```suggestion
    await c.submit(slowinc, 1, pure=False, delay=0.1)
    while not len(p.source.data["left"]):
```
Is resubmitting necessary?

Reply:
Yeah, if the profiler didn't run during the task then we're screwed I think.  We want to continuously run stuff until the profiler picks it up and that information arrives on the scheduler.

Reply:
Makes sense, thanks